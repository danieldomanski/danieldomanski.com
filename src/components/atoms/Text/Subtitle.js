import React from 'react'
import Text from '.'

const Subtitle = ({ children, ...rest }) => (
  <Text
    as="h3"
    display="block"
    fontSize={['lg', 'lg', 'xl', 'xl']}
    fontWeight="black"
    fontColor="primary.5"
    mb={[6, 8, 12, 12, 16]}
    flex={1}
    style={{ textTransform: 'uppercase' }}
    {...rest}
  >
    {children}
  </Text>
)

export default Subtitle
