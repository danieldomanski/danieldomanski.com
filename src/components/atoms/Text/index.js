import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import tw from 'tailwind.macro'
import { layout, typography, color, space, flexbox } from 'styled-system'

const TextContainer = styled.p`
  ${tw`h-auto`};

  ${layout};
  ${typography};
  ${color};
  ${space};
  ${flexbox};

  &:hover {
    color: ${props => (props.hover ? props.hover.color : null)};
  }

  transition: color 0.1s ease-in-out;
`

const Text = ({
  children,
  display,
  fontFamily,
  fontSize,
  fontWeight,
  fontColor,
  lineHeight,
  maxWidth,
  hover,
  mt,
  mb,
  mr,
  ml,
  ...rest
}) => (
  <TextContainer
    display={display}
    fontSize={fontSize}
    fontFamily={fontFamily}
    fontWeight={fontWeight}
    color={fontColor}
    lineHeight={lineHeight}
    maxWidth={maxWidth}
    hover={hover}
    mt={mt}
    mb={mb}
    mr={mr}
    ml={ml}
    {...rest}
  >
    {children}
  </TextContainer>
)

Text.propTypes = {
  fontSize: PropTypes.oneOfType([PropTypes.string, PropTypes.array]),
  fontWeight: PropTypes.oneOf(['normal', 'bold', 'black']),
  fontFamily: PropTypes.oneOf(['sans', 'serif', 'mono']),
  lineHeight: PropTypes.oneOf(['tight', 'normal', 'relaxed', 'loose']),
  fontColor: PropTypes.string,
  children: PropTypes.node.isRequired,
  display: PropTypes.string,
  mb: PropTypes.oneOfType([PropTypes.number, PropTypes.array]),
  mt: PropTypes.oneOfType([PropTypes.number, PropTypes.array]),
  mr: PropTypes.oneOfType([PropTypes.number, PropTypes.array]),
  ml: PropTypes.oneOfType([PropTypes.number, PropTypes.array]),
}

Text.defaultProps = {
  fontFamily: 'serif',
  fontSize: ['4xl'],
  lineHeight: 'normal',
  fontWeight: 'normal',
  fontColor: 'primary.5',
  display: ['inline-block'],
  mb: 0,
  mt: 0,
  mr: 0,
  ml: 0,
}

export default Text
