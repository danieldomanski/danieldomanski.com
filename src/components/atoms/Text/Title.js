import React from 'react'
import Text from './index'

const Title = ({ children }) => (
  <Text
    fontFamily={'sans'}
    fontSize={['3xl', '4xl', '4xl', '5xl']}
    lineHeight={'normal'}
    fontWeight={'black'}
    fontColor={'primary.11'}
    display={'block'}
    textAlign="center"
    mt={[8, 8, 8, 12, 16]}
    mb={[8, 8, 8, 12, 16]}
    style={{ letterSpacing: '-0.04em' }}
  >
    {children}
  </Text>
)

export default Title
