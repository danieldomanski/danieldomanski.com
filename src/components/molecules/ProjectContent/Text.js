import React from 'react'
import Text from '../../atoms/Text'

export const ImageCaptionTitle = ({ children }) => (
  <Text
    fontWeight="black"
    fontSize={['3xl', '3xl', '4xl']}
    fontColor="primary.11"
    lineHeight="tight"
    mb={8}
  >
    {children}
  </Text>
)

export const ImageCaptionDescription = ({ children }) => (
  <Text
    fontColor="primary.8"
    fontSize={['lg', 'lg', 'lg', 'xl']}
    fontWeight="medium"
  >
    {children}
  </Text>
)
