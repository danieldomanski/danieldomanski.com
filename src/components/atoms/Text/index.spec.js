import 'jest-styled-components'
import '@testing-library/jest-dom/extend-expect'
import React from 'react'
import { render, cleanup } from '@testing-library/react'

import Text from './index'

afterEach(cleanup)

test('<Text> renders with different styled-system category props', () => {
  const { container } = render(
    <Text fontWeight="bold" fontColor={['red', 'blue']} fontSize={['lg', 'xl']}>
      Test.
    </Text>
  )

  expect(container.firstChild).toMatchSnapshot()
})
