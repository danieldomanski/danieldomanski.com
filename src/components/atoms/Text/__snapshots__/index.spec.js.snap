// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text> renders with different styled-system category props 1`] = `
.c0 {
  display: inline-block;
  font-size: lg;
  font-family: sans;
  font-weight: bold;
  line-height: normal;
  color: red;
  margin-top: 0;
  margin-bottom: 0;
  margin-right: 0;
  margin-left: 0;
  height: auto;
  -webkit-transition: color 0.1s ease-in-out;
  transition: color 0.1s ease-in-out;
}

@media screen and (min-width:40em) {
  .c0 {
    font-size: xl;
  }
}

@media screen and (min-width:40em) {
  .c0 {
    color: blue;
  }
}

<p
  class="c0"
  color="red,blue"
  display="inline-block"
  font-family="sans"
  font-size="lg,xl"
  font-weight="bold"
>
  Test.
</p>
`;
